src/sage/rings/polynomial/pbori/pbori.pyx

```diff
@@ -4105,6 +4106,100 @@ cdef class BooleanPolynomial(MPolynomial):
         """
         return new_BS_from_PBSet(self._pbpoly.set(), self._parent)
 
+    def separate(self):
+        r"""
+        Returns a list of polynomials whose sum is ``self`` and
+        whose sets of variables are mutually disjoint
+
+
+        EXAMPLES::
+
+            sage: B.<x,y,z,w> = BooleanPolynomialRing(4)
+            sage: (x*y+x+z*w).separate()
+            [x*y + x, z*w]
+        """
+        if self.nvariables() <= 1:
+            return [self]
+        fp = (self/self.variable(0)) * self.variable(0)
+        fq = self.subs({self.variable(0): 0})
```

Comment:
I think it's more efficient to just write `fq = self - fp`?

Reply:
We've tested both of them, and in most cases the original one is about 10% faster than `fq = self - fp`.